server.js

```diff
@@ -66,6 +67,20 @@ app.put('/api/foods/:id', function(request, response) {
   })
 })
 
+app.get('/api/diary', function(request, response) {
```

Comment:
hey y'all! sorry for the late reply. i like @lao9's suggestion. i think breaking this endpoint down will give you more flexibility and control over what data is returned when.

you're probably running into a scope problem. i'm not seeing anything specific when looking at your code by looking at it here. i'm assuming where you have the pry, it's constructed as you'd hoped/expected and then when you're returning the json on line 80, it no longer contains the data you need?